plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/SingletonJdbcIdentityCacheMapping.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+public class SingletonJdbcIdentityCacheMapping
+        implements JdbcIdentityCacheMapping
+{
+    @Override
+    public JdbcIdentityCacheKey getRemoteUserCacheKey(JdbcIdentity identity)
+    {
+        return SingletonJdbcIdentityCacheKey.INSTANCE;
+    }
+
+    private static final class SingletonJdbcIdentityCacheKey
+            extends JdbcIdentityCacheKey
+    {
+        private static final SingletonJdbcIdentityCacheKey INSTANCE = new SingletonJdbcIdentityCacheKey();
+
+        @Override
+        public int hashCode()
+        {
+            return 0;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            return obj == INSTANCE;
```

Comment:
e.g. if JdbcIdentityCacheKey becomes serializable, this would need to become `obj instanceof SingletonJdbcIdentityCacheKey`. IMO it's safer to write it that way even now